src/foam/nanos/fs/FileDataDAO.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.fs',
+  name: 'FileDataDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  javaImports: [
+    'foam.blob.*'
+  ],
+
+  properties: [
+    {
+      class: 'Long',
+      name: 'maxStringDataSize',
+      value: 1024 * 3
+    }
+  ],
+
+  methods: [
+    {
+      documentation: `Server side logic to save file to either file.dataString or to physical files system.`,
+      name: 'put_',
+      javaCode: `
+      File file = (File) obj;
+      if ( ! foam.util.SafetyUtil.isEmpty(file.getDataString()) ) {
+        return getDelegate().put_(x, obj);
+      }
+
+      Blob blob = file.getData();
+      if ( blob == null ) {
+        return getDelegate().put_(x, obj);
+      }
+
+      // save to data string
+      if ( file.getFilesize() <= getMaxStringDataSize() ) {
+        try {
+          java.io.ByteArrayOutputStream os = new java.io.ByteArrayOutputStream((int) file.getFilesize());
+          blob.read(os, 0, file.getFilesize());
+          String encodedString = java.util.Base64.getEncoder().encodeToString(os.toByteArray());
+          String type = file.getMimeType();
+          if ( foam.util.SafetyUtil.isEmpty(type) ) {
+            if ( ! foam.util.SafetyUtil.isEmpty(file.getFilename()) &&
+                 file.getFilename().lastIndexOf(".") != -1 &&
+                 file.getFilename().lastIndexOf(".") != 0 ) {
+              type = "text/" + file.getFilename().substring(file.getFilename().lastIndexOf(".")+1);
```

Comment:
Should we have a map of extensions and mine types? https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types. So that for .txt file the type is "text/plain" instead of "text/txt".

Reply:
@lchanmann Yeah, we have the start of it in fs/SupportFileTypeDAO and fs/FileType . 
The above is just copy of what we currently have. 
So a future ticket.